### PR TITLE
docs: update local rate limit with descriptors example

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

Commit Message: The example is incorrect since local rate limit does not implement virtual host fallback
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
Issue: https://github.com/envoyproxy/envoy/issues/14851

### CODE REVIEW

**Comment by phlax** [comment]
@kyessenov can we move this config to a `literalinclude` yaml file please

it means it would get tested in ci

**Comment by kyessenov** [comment]
@phlax Unfortunately, the old version is a valid YAML but it is not implemented, so it would not catch the issue.

**Comment by kyessenov** [comment]
Actually, it's already a validated code block. Is it really necessary to split it away?

**Comment by phlax** [comment]
> Is it really necessary to split it away?

kinda, yes

using a literalinclude means it gets validated with a config load check in addition to the yaml validation

it also makes it much easier to maintain and edit etc - and it means users can download a complete working example

**Comment by kyessenov** [comment]
I'm confused - do we need to specify the entire envoy bootstrap for this? I can't find examples of a standalone HCM.

**Comment by phlax** [comment]
> do we need to specify the entire envoy bootstrap for this?

yep - but you can specify which lines to show (and also to emphasize if you want)

**Comment by kyessenov** [comment]
@phlax PTAL. BTW the existing configs have issues, `advanced/matching/_include/request_response.yaml` is missing connect timeout.

**Comment by phlax** [comment]
>  BTW the existing configs have issues, advanced/matching/_include/request_response.yaml is missing connect timeout.

`connect_timeout` now has a default of 5s and is not required - i have been going through and removing from any examples where its not the focus of the example

**Comment by kyessenov** [comment]
Ah, OK. I must have an old build hanging around.

**Comment by phlax** [comment]
@kyessenov windows ci is broken atm - unrelated to your pr

File: docs/root/configuration/http/http_filters/_include/local-rate-limit-with-descriptors.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
+static_resources:
+  listeners:
+    - address:
+        socket_address:
+          address: 0.0.0.0
+          port_value: 8000
+      listener_filters:
```
Comment (phlax) [inline docs/root/configuration/http/http_filters/_include/local-rate-limit-with-descriptors.yaml:7]: is this required ?

File: docs/root/configuration/http/http_filters/_include/local-rate-limit-with-descriptors.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+              - endpoint:
+                  address:
+                    socket_address:
+                      address: 127.0.0.1
+                      port_value: 8080
+    - name: service_protected_by_rate_limit
+      load_assignment:
+        cluster_name: service_protected_by_rate_limit
+        endpoints:
+          - lb_endpoints:
```
Comment (phlax) [inline docs/root/configuration/http/http_filters/_include/local-rate-limit-with-descriptors.yaml:92]: not a blocker - but the mixed use of indenting styles for arrays can make config harder to grok

im going to clean these up at some point (pending pr is here #15726 ) 

generally least indent is preferred

File: docs/root/configuration/http/http_filters/local_rate_limit_filter.rst [hunk 3]
```diff
@@ -127,69 +127,9 @@ the default token bucket is used.
...
-      - actions: # any actions in here
-        - request_headers:
-            header_name: x-envoy-downstream-service-cluster
-            descriptor_key: client_cluster
-        - request_headers:
-            header_name: ":path"
-            descriptor_key: path
+.. literalinclude:: _include/local-rate-limit-with-descriptors.yaml
+   :language: yaml
+   :lines: 16-76
```
Comment (phlax) [inline docs/root/configuration/http/http_filters/local_rate_limit_filter.rst:132]: could we add a download link - ie:

```
:caption: :download:`local-rate-limit-with-descriptors.yaml < _include/local-rate-limit-with-descriptors.yaml>`
```

**Review by phlax — CHANGES_REQUESTED** [review]
looking good @kyessenov - couple of nits below

File: docs/root/configuration/http/http_filters/_include/local-rate-limit-with-descriptors.yaml (hunk 1 above)
Comment (kyessenov) [reply docs/root/configuration/http/http_filters/_include/local-rate-limit-with-descriptors.yaml:7]: No, it's not. Removed.

File: docs/root/configuration/http/http_filters/local_rate_limit_filter.rst (hunk 3 above)
Comment (kyessenov) [reply docs/root/configuration/http/http_filters/local_rate_limit_filter.rst:132]: Done.

File: docs/root/configuration/http/http_filters/_include/local-rate-limit-with-descriptors.yaml (hunk 2 above)
Comment (kyessenov) [reply docs/root/configuration/http/http_filters/_include/local-rate-limit-with-descriptors.yaml:92]: Done.

**Review by phlax — APPROVED** [review]
great! thanks @kyessenov - lgtm

**Pull request merged**